language_model/ngram_language_model.py

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+########################################################################
+# 
+# Copyright (c) 2017 Baidu.com, Inc. All Rights Reserved
+# 
+########################################################################
+"""
+File: ngram_language_model.py
+Author: chengxiaohua(chengxiaohua@baidu.com)
+Date: 2017/05/02 15:13:21
+"""
+import sqlite3
+import paddle.v2 as paddle
+import math
+import os
+import sys
+import ptb
+import gzip
+import numpy as np
+
+batch_size = 1000
+word_dim = 256
+hidden_dim = 512
+lr = 0.1
+max_word_num = 40
+
+
+def gen_embedding_layer(layer_name, voc_dim):
+    return paddle.layer.data(
+        name=layer_name, type=paddle.data_type.integer_value(voc_dim))
```

Comment:
`gen_embedding_layer` is not necessary and the name is confusing. I suggest to directly call the data layer.